### DISCUSSION TITLE
Omit Newtonsoft's specific `$type` property in favor of JsonConverter

### DISCUSSION THREAD

**Discussion by mcrio** [ravendb/ravendb#15589] [Q&A]
Hi,

When a document holds a property of a base abstract type, the RavenDb serializer automatically adds the `$type` property in order to be able to deserialize to the proper derived type.

I'd rather not rely on this feature but would like to have a custom `JsonConverter` so I don't have to worry the namespace changes along the way and breaks deserialization of existing items. By not having the `$type` we'd not miss writing the converter.

What is the correct way to disable this feature and are the other implications of that action?

As I can see it could be disabled by doing: 
```
var serializerConventions = new NewtonsoftJsonSerializationConventions();
serializerConventions.CustomizeJsonSerializer = serializer =>
{
    serializer.TypeNameHandling = TypeNameHandling.None;
};
serializerConventions.CustomizeJsonDeserializer = serializer =>
{
    serializer.Converters.Add(new MyCustomConverter());
};
```

**Comment by ayende** [comment]
Easiest would be to simply place `[JsonConverter(typeof(MyPolymorphicConverter)]` on the base class, I think.

**Reply by mcrio** [reply]
Yes but looking for a way to add it through configuration as the assembly with the persistence model does not depend on Newtonsoft.

Do you see any issues handling it like in the example above. The tests are green for now : )

**Reply by ayende** [reply]
Yes, that can work
